tests/test_examples.py

```diff
@@ -1423,3 +1466,45 @@ def test_regressions(tmp_path, request):
             check_subprocess=True,
             uninstall=True,
         )
+
+
+@pytest.mark.parametrize("no_registry", (0, 1))
+@pytest.mark.skipif(not ON_CI, reason="CI only")
+@pytest.mark.skipif(not sys.platform.startswith("win"), reason="Windows only")
+def test_not_in_installed_menu_list_(tmp_path, request, no_registry):
+    """Verify the app is Installed Apps Menu (or not), based the CLI arg '/NoRegistry'.
+    If NoRegistry=0, we expect to find the installer in the Menu, otherwise not.
+    """
+    input_path = _example_path("extra_files")  # The specific example we use here is not important
+    options = ["/InstallationType=JustMe", f"/NoRegistry={no_registry}"]
+    for installer, install_dir in create_installer(input_path, tmp_path):
+        _run_installer(
+            input_path,
+            installer,
+            install_dir,
+            request=request,
+            check_subprocess=True,
+            uninstall=False,
+            options=options,
+        )
+
+    # Extract name and version from yaml-file
+    # in order to predict parts of the installer name
+    with open(input_path / "construct.yaml") as f:
+        contents = f.readlines()
+    name = ""
+    version = ""
+    for line in contents:
+        if "name: " in line:
+            name = line.split(":")[-1].strip()
+        if "version: " in line:
+            version = line.split(":")[-1].strip()
+        if name and version:
+            break
+    partial_name = f"{name} {version}"
```

Comment:
This seems brittle. Instead of manually parsing the `construct.yaml` file, I suggest using the `UninstallString` value and see if it starts with `$INSTDIR\Uninstall-`: https://github.com/conda/constructor/blob/7a837bd9d3c94813cc4734ebc6f9879557ddd488/constructor/nsis/main.nsi.tmpl#L1627-L1628